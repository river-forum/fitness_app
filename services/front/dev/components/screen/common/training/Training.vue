<template>
  <div class="label-wrapper">
    <div class="flex">
      <RunLabel :current="current" @selected="selectedListener" />
      <MachineLabel :current="current" @selected="selectedListener" />
      <StretchLabel :current="current" @selected="selectedListener" />
      <WeightLabel :current="current" @selected="selectedListener" />
    </div>
    <component :is="currentComponent" class="mt-10" />
  </div>
</template>

<script>
import { defineComponent, ref, computed } from '@nuxtjs/composition-api'
import RunLabel from './label/Run'
import MachineLabel from './label/Machine'
import StretchLabel from './label/Stretch'
import WeightLabel from './label/Weight'
import RunInner from './contents/Run'
import MachineInner from './contents/Machine'
import StretchInner from './contents/Stretch'
import WeightInner from './contents/Weight'

export default defineComponent({
  components: {
    RunLabel,
    MachineLabel,
    StretchLabel,
    WeightLabel,
    RunInner,
    MachineInner,
    StretchInner,
    WeightInner,
  },
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  setup(props, ctx) {
    const current = ref('run')

    function selectedListener(val) {
      current.value = val
      ctx.emit('current', val)
    }

    const currentComponent = computed(() => {
      switch (current.value) {
        case 'run':
          return RunInner
        case 'machine':
          return MachineInner
        case 'stretch':
          return StretchInner
        case 'weight':
          return WeightInner
      }
    })

    return {
      current,
      selectedListener,
      currentComponent,
    }
  },
})
</script>
