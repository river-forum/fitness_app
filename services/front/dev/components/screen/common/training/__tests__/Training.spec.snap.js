import { mount } from '@vue/test-utils'
import Training from '../Training.vue'

it('Training Component renders correctly', () => {
  const app = mount(Training)

  expect(app.html()).toMatchSnapshot()
})
