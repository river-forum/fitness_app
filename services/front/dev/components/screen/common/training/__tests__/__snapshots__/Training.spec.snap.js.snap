// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training Component renders correctly 1`] = `
"<div class=\\"label-wrapper\\">
  <div class=\\"flex\\">
    <div class=\\"w-1/4 text-white label\\"><button class=\\"relative flex items-center justify-center w-full bg-gray-400 label__body\\" style=\\"background-color: rgb(211, 125, 113);\\">
        <div class=\\"label__item\\">Run</div>
        <div class=\\"absolute left-0 right-0 m-auto text-center triangle-wrap\\"><span class=\\"triangle-wrap__item\\" style=\\"border-color: #d37d71 transparent transparent transparent;\\"></span></div>
      </button></div>
    <div class=\\"w-1/4 text-white label\\"><button class=\\"relative flex items-center justify-center w-full bg-gray-400 label__body\\" style=\\"background-color: rgb(113, 142, 197);\\">
        <div class=\\"label__item\\">マシーン</div>
        <!---->
      </button></div>
    <div class=\\"w-1/4 text-white label\\"><button class=\\"relative flex items-center justify-center w-full bg-gray-400 label__body\\" style=\\"background-color: rgb(161, 197, 86);\\">
        <div class=\\"label__item\\">ストレッチ</div>
        <!---->
      </button></div>
    <div class=\\"w-1/4 text-white label\\"><button class=\\"relative flex items-center justify-center w-full bg-gray-400 label__body\\" style=\\"background-color: rgb(246, 221, 114);\\">
        <div class=\\"label__item\\">Weight</div>
        <!---->
      </button></div>
  </div>
  <div class=\\"test mt-10\\">
    <inputdate></inputdate>
    <togglebtn button-inner-texts=\\"RUN,BIKE\\" class=\\"mt-12\\"></togglebtn>
    <inputnumber type=\\"distance\\" class=\\"mt-12\\"></inputnumber> <button button-inner-text=\\"記録する\\" class=\\"mx-auto mt-16\\"></button>
  </div>
</div>"
`;
